tests/model_fields/test_decimalfield.py

```diff
@@ -91,7 +91,10 @@ def test_lookup_really_big_value(self):
         Really big values can be used in a filter statement.
         """
         # This should not crash.
-        Foo.objects.filter(d__gte=100000000000)
+        list(Foo.objects.filter(d__gte=100000000000))
```

Comment:
I noticed this wasn't being evaluated and wondered what it's purpose was… after evaluating I observed it started failing with the same `decimal.InvalidOperation` exception.

Reply:
Could we instead use some assertion to make the test a bit more... test-ish? :-)

Something like:

```suggestion
       # Ensure that the query is executed
       self.assertEqual(list(Foo.objects.filter(d__gte=100000000000)), [])
```

Reply:
(In this case we don't need to wrap it)

Reply:
TIL, thank you! :star_struck:

Reply:
For some context, the test was introduced in e9814029f570bd0866dc859147bca90340bcc913 to address ticket-10692 but since the queryset wasn't evaluated I think it was broken from the beginning.

Unless it was changed during the expressions refector, which I doubt in this case, `get_db_prep_value` and `get_db_prep_save` method are only called at queryset compilation time (see `Lookup.as_sql`) and not at resolving time (`QuerySet.filter`).